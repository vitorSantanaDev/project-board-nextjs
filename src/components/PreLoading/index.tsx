import * as S from './styles'

const PreLoading: React.FC = () => {
  return (
    <S.Wrapper>
      <S.Logo
        src="/images/logo.svg"
        alt="Uma logo com uma letra 'B' e um círculo amarelo ao lado"
      />
      <S.Title>A simple Project Board in NextJS</S.Title>
      <S.Description>
        TypeScript, ReactJS, NextJS and Styled components
      </S.Description>
      <S.Illustration
        src="/images/hero-illustration.svg"
        alt="Um usuário de tecnologia, de frente pra um notebook."
      />
    </S.Wrapper>
  )
}

export default PreLoading
