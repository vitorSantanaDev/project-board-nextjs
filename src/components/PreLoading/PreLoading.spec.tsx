import { render, screen } from '@testing-library/react'
import PreLoading from '.'

describe('<PreLoading />', () => {
  it('should render two heading and two images', () => {
    render(<PreLoading />)
    const firstImage = screen.getByRole('img', {
      name: /uma logo com uma letra 'b' e um círculo amarelo ao lado/i
    })
    const secondImage = screen.getByRole('img', {
      name: /um usuário de tecnologia, de frente pra um notebook\./i
    })
    const firstHeading = screen.getByRole('heading', {
      name: /a simple project board in nextjs/i
    })
    const secondHeading = screen.getByRole('heading', {
      name: /a simple project board in nextjs/i
    })
    expect(firstImage).toBeInTheDocument()
    expect(secondImage).toBeInTheDocument()
    expect(firstHeading).toBeInTheDocument()
    expect(secondHeading).toBeInTheDocument()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<PreLoading />)
    expect(container.firstChild).toHaveStyle({ 'background-color': '#17181f' })
  })

  it('shoul match snapshot', () => {
    const { container } = render(<PreLoading />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
