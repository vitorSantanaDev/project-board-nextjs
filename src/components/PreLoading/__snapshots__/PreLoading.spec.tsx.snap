// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PreLoading /> shoul match snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  color: #fff;
  background-color: #17181f;
  text-align: center;
  padding: 3rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Uma logo com uma letra 'B' e um círculo amarelo ao lado"
    class="c1"
    src="/images/logo.svg"
  />
  <h1
    class="c2"
  >
    A simple Project Board in NextJS
  </h1>
  <h2
    class="c3"
  >
    TypeScript, ReactJS, NextJS and Styled components
  </h2>
  <img
    alt="Um usuário de tecnologia, de frente pra um notebook."
    class="c4"
    src="/images/hero-illustration.svg"
  />
</main>
`;
